Sources/Extensions/SwiftStdlib/IntExtensions.swift

```diff
@@ -171,7 +171,13 @@ public extension Int {
 		return n == 0 ? self : Int(round(Double(self) / Double(n))) * n
 	}
 	// swiftlint:enable identifier_name
-
+    
+    /// SwifterSwift: Array of factors for given integer.
+    /// - Returns: array of factors
+    public func factors() -> [Int] {
+        let factors = (1...self).filter { self % $0 == 0 }
```

Comment:
There is actually a more efficient way to get this. By checking [this SO question](https://stackoverflow.com/questions/6800193/what-is-the-most-efficient-way-of-finding-all-the-factors-of-a-number-in-python) we can convert the answer to:

```
func factors() -> [Int] {
    guard self > 0 else { return [] }
    return [Int]((1...Int(ceil(sqrt(Double(self)))) + 1)
        .reduce(Set<Int>(), { (result, val) -> Set<Int> in
            guard self % val == 0 else { return result }
            return result.union([val, self / val])
        }))
        .sorted()
}
```

You don't have to sort it, but I would imagine that the user would expect it to be so.

Reply:
Hmm, if you want you can make it generic for all `BinaryIntegers` (`UInt`, `Int8`, etc.), but it's a bit ugly.

```
extension BinaryInteger {
    func factors() -> [Self] {
        guard self > 0 else { return [] }
        return [Self]((1...Int64(Double(Int64(self)).squareRoot()) + 1)
            .reduce(Set<Self>(), { (result, val) -> Set<Self> in
                let val: Self = numericCast(val)
                guard self % val == 0 else { return result }
                return result.union([val, self / val])
            }))
            .sorted()
    }
}
```

Reply:
Hey @n0an @guykogus 
@guykogus I like your proposal and in fact it is more efficient
But maybe we can have as an option, instead of using `reduce` we use `reduce(into:)` to avoid  copy and use in-place mutation :))
Or even use a for loop implementation like:
```swift
func factors() -> [Int] {
        guard
            self > 0 else { return [] }
        
        let rootSquared = Int(Double(self).squareRoot())
        var factors = Set<Int>()
        for i in 1...(rootSquared + 1) where self % i == 0 {
            factors.insert(i)
            factors.insert(self/i)
        }
        return factors.sorted()
}
```
To me it seems more simpler :)) 
Let me know what you guys think?

Reply:
Stop doing that stupid thing with the `guard` statement @LucianoPAlmeida 🤦‍♂️ 😭 😭

Reply:
@SD10 I can't help myself, it is stronger than me 😂😂

Reply:
@LucianoPAlmeida I think I prefer your for-loop method, although I'd still use `factors.formUnion([val, self / val])` instead of `insert`ing twice.
Also, what's the "stupid thing with the `guard` statement"? 🤔

Reply:
@guykogus , @LucianoPAlmeida , @SD10 , I like suggested solution. As for `guard` I suppose this is about new line and indenting :)

Reply:
Yeap is about the new line indenting  😂😂
@guykogus About the `fromUnion` I agree :)) I just did with the `insert` because was just two calls and I wanted to avoid the array allocation to call the method, but I think is not a problem.

Reply:
Anyone want to take a crack at updating this PR? 😬